x-pack/examples/testing_embedded_lens/public/app.tsx

```diff
@@ -0,0 +1,679 @@
+    },
+    {
+      id: 'area',
+      attributes: getLensAttributes(props.defaultDataView, fields, 'area', 'green'),
+    },
+    { id: 'pie', attributes: getLensAttributesPartition(props.defaultDataView, fields) },
+    { id: 'table', attributes: getLensAttributesDatatable(props.defaultDataView, fields) },
+    { id: 'heatmap', attributes: getLensAttributesHeatmap(props.defaultDataView, fields) },
+    { id: 'gauge', attributes: getLensAttributesGauge(props.defaultDataView, fields) },
+  ];
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const charts = useMemo(() => [...defaultCharts, ...loadedCharts], [loadedCharts]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialAttributes = useMemo(() => JSON.stringify(charts[0].attributes, null, 2), []);
+
+  const [currentSO, saveSO] = useState(initialAttributes);
+  const switchChartPreset = useCallback(
+    (newIndex) => {
+      const newChart = charts[newIndex];
+      saveSO(JSON.stringify(newChart.attributes, null, 2));
+    },
+    [charts]
+  );
+
+  const currentAttributes = useMemo(() => JSON.parse(currentSO), [currentSO]);
+
+  const isDisabled = !currentAttributes;
+  const isColorDisabled = isDisabled || !isXYChart(currentAttributes);
+
+  return (
+    <KibanaContextProvider services={{ uiSettings: props.core.uiSettings }}>
+      <EuiPageTemplate fullHeight template="empty">
+        <EuiFlexGroup
+          className="eui-fullHeight"
+          gutterSize="none"
+          direction="column"
+          responsive={false}
+        >
+          <EuiFlexItem className="eui-fullHeight">
+            <EuiFlexGroup className="eui-fullHeight" gutterSize="l">
+              <EuiFlexItem grow={3}>
+                <EuiPanel hasShadow={false}>
+                  <p>
+                    This app embeds a Lens visualization by specifying the configuration. Data
+                    fetching and rendering is completely managed by Lens itself.
+                  </p>
+                  <p>
+                    The editor on the right hand side make it possible to paste a Lens attributes
+                    configuration, and have it rendered. Presets are available to have a starting
+                    configuration, and new presets can be saved as well (not persisted).
+                  </p>
+                  <p>
+                    The Open with Lens button will take the current configuration and navigate to a
+                    prefilled editor.
+                  </p>
+                  <EuiSpacer />
+                  <EuiFlexGroup wrap>
+                    <EuiFlexItem grow={false}>
+                      <EuiButton
+                        data-test-subj="lns-example-change-color"
+                        isLoading={isLoading}
+                        onClick={() => {
+                          // eslint-disable-next-line no-bitwise
+                          const newColor = '#' + ((Math.random() * 0xffffff) << 0).toString(16);
```

Comment:
nit: you can also do the below to avoid the bitwise eslint warning :) 
```suggestion
                          const newColor = `#${Math.floor(Math.random()*16777215).toString(16)}`;
```